node/bft/src/helpers/pending.rs

```diff
@@ -105,10 +105,10 @@ impl<T: Copy + Clone + PartialEq + Eq + Hash, V: Clone> Pending<T, V> {
         // Clear the callbacks that have expired.
         self.clear_expired_callbacks_for_item(item);
         // Return the number of live callbacks.
-        self.callbacks
-            .lock()
+        self.pending
+            .write()
             .get(&item)
-            .map_or(0, |callbacks| callbacks.iter().filter(|(_, _, request_sent)| *request_sent).count())
+            .map_or(0, |peers| peers.values().flatten().filter(|(_, _, request_sent)| *request_sent).count())
```

Comment:
Since `num_sent_requests()` will be called very frequently by the `Worker::send_transmission_request()` and iterating nested hashmaps isn't constant-time, it might make sense to consider memoizing these counts in `Pending`-struct.

Reply:
In addition, if we do need a linear lookup here, we might want to call `.clone()` after the `.get` in order to release the lock sooner

Reply:
I believe since we are now clearing the callbacks after expiration, the size of this vec shouldn't grow too large.

I'd like to keep the change-set small enough s.t. it is easy to review, but would be open to further optimization PRs down the line.